metabase/metabase
# Refactor before fix: Retrieve multiple get params for Public Question

This PR is a refactor in preparation for #16795 — please see discussion below https://github.com/metabase/metabase/pull/16765#discussion_r659088975

Related to bug #7120

## How to Test

1. Enable public sharing in admin.
2. Ask a question
3. Native question
4. Sample Dataset
5. `SELECT * FROM PEOPLE WHERE {{birthdate}} AND {{source}} AND STATE={{state}}`
6. `{{birthdate}}` as Field Filter of People.BirthDate with default of `Past 30 Years`
`{{source}}` as Field Filter of People.Source with default of `Affiliate`
`{{state}}` as Text filter with Required enabled and default of `CA`

![image](https://user-images.githubusercontent.com/1447303/85953660-63343e80-b972-11ea-8846-fdc714cda796.png)

7. Click on share icon at bottom right.
8. Copy url from top input or copy button
9. Visit that url, which should end with a uuid.

URL shoud **NOT YET** be automatically appended with `?birthdate=past30years&source=Affiliate&state=CA`
